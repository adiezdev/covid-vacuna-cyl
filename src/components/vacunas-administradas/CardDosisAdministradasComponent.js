import React from 'react'
import PropTypes from 'prop-types'
import { SpinnerComponent } from '../SpinnerComponent'
/**
 * 
 * @param {*} param0 
 * @returns 
 * 
 * Component card vaccine
 */
export const CardDosisAdministradasComponent = ({ total,title, img, loading}) => {


    return (
        <div className="maincard card-fiexWidth-midle">
            <header>
                <img 
                className='header-img'
                src={`${img}.svg`} 
                alt='icono_vacuna' 
                />
                <p className='t-midle'>{ title }</p>
            </header>
            {
                !loading
                ?
                <div className="data t-big">{total}</div>
                :
                <SpinnerComponent></SpinnerComponent>
            }
        </div>
    )
}

/**
 * Proptypes
 * To form the card, the title and the total are necessary
 */
CardDosisAdministradasComponent.propTypes= {
    total: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired
}