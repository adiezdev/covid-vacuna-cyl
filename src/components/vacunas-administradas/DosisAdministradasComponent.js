import React from 'react'
import { CardDosisAdministradasComponent } from './CardDosisAdministradasComponent'
import { useFetch } from '../../hooks/useFetch'
import { getPersonasVacunadas } from '../../selectors/personas-vacunadas/getPersonasVacunadas'

/**
 * Component form cards 
 *  -vacunas administradas
 *  -ciclo completo
 * @returns 
 */
export const DosisAdministradasComponent = () => {

    const { data , isLogin } = useFetch( 1000 ,  'personas-vacunadas-covid')
    const { total , ciclototal  } = getPersonasVacunadas(data)
    //Create array to tow data 
    const informations = [
        {
            id:  0,
            title:'Personas vacunadas',
            sumaReduce: total,
            img: './assets/icons/admin_vacuna'
        },
        {
            id:  1,
            title:'Personas con pauta completa',
            sumaReduce: ciclototal,
            img: './assets/icons/vacuna_completa'
        }
    ]
    return (
        <>
            <div className="center-item-wrap flex-column">
                {
                    informations.map( information => (
                        <CardDosisAdministradasComponent
                            key={information.id}
                            total={information.sumaReduce}
                            title={information.title}
                            img={information.img}
                            loading={isLogin}
                        />
                    ))
                }
            </div>
        </>
        
    )
}