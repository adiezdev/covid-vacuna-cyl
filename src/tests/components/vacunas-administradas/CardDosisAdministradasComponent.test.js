import Enzyme , { shallow } from "enzyme";
import { shallowToJson } from 'enzyme-to-json'

import { CardDosisAdministradasComponent } from "../../../components/vacunas-administradas/CardDosisAdministradasComponent";
import { informations } from "../../fixtures/data";

import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Pruebas unitarias para <CardDosisAdministradasComponent/>',() =>{

    const wrapper = shallow( 
    <CardDosisAdministradasComponent 
        total={informations[0].sumaReduce} 
        title={informations[0].title} 
        img={informations[0].img}
        loading={false}/>)

        
    test('Renderizar correctamente el componente',()=>{
        expect( shallowToJson(wrapper) ).toMatchSnapshot()
    })
   
    test('Debe de mostrar el total correctamente', ()=>{

        const div = wrapper.find('.data')
        expect(div.text()).toBe(`${informations[0].sumaReduce}`)
    })

    test('No debe de cargar el total y debe de mostrarse el spinner',()=>{

        const wrapper = shallow(
            <CardDosisAdministradasComponent 
        total={informations[0].sumaReduce} 
        title={informations[0].title} 
        img={informations[0].img}
        loading={true}/>)

        expect(wrapper.find('SpinnerComponent').length ).toBe( 1 )
    })
})