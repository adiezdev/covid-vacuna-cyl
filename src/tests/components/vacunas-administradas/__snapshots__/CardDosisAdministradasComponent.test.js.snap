// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias para <CardDosisAdministradasComponent/> Renderizar correctamente el componente 1`] = `
<div
  className="maincard card-fiexWidth-midle"
>
  <header>
    <img
      alt="icono_vacuna"
      className="header-img"
      src="./assets/icons/admin_vacuna.svg"
    />
    <p
      className="t-midle"
    >
      Personas vacunadas
    </p>
  </header>
  <div
    className="data t-big"
  >
    10000000
  </div>
</div>
`;
